Alastair McKinstry <mckinstry@debian.org>
Description: Use binaries from install location, which is where they are
 during the Debian build process.
Last-Updated: 2015-10-25
Forwarded: not-needed

Index: emoslib-4.4.9/tests/bufr/CMakeLists.txt
===================================================================
--- emoslib-4.4.9.orig/tests/bufr/CMakeLists.txt
+++ emoslib-4.4.9/tests/bufr/CMakeLists.txt
@@ -20,14 +20,14 @@ endif()
 # FIXME: add PROPERTIES to ecbuild_add_test()
 ecbuild_add_test(
     TARGET       bufr_decode_all_NotFoundTestFile
-    COMMAND      ${CMAKE_BINARY_DIR}/bin/bufr_decode_all
+    COMMAND      ${CMAKE_INSTALL_PREFIX}/bin/bufr_decode_all
     ARGS         -i "NotFoundTestFile.bufr"
     ENVIRONMENT  "BUFR_TABLES=${TEST_BUFR_TABLES_PATH}" )
 set_tests_properties( bufr_decode_all_NotFoundTestFile PROPERTIES WILL_FAIL TRUE )
 
 ecbuild_add_test(
     TARGET       bufr_decode_all_NotFoundTablesPath
-    COMMAND      ${CMAKE_BINARY_DIR}/bin/bufr_decode_all
+    COMMAND      ${CMAKE_INSTALL_PREFIX}/bin/bufr_decode_all
     ARGS         -i ${PROJECT_SOURCE_DIR}/tests/bufr/data/temp_101.bufr
     ENVIRONMENT  "BUFR_TABLES=NotFoundTablesPath/" )
 set_tests_properties( bufr_decode_all_NotFoundTablesPath PROPERTIES WILL_FAIL TRUE )
@@ -38,7 +38,7 @@ set_tests_properties( bufr_decode_all_No
 set( data ${PROJECT_SOURCE_DIR}/tests/bufr/data )
 ecbuild_add_test(
     TARGET       emoslib_bufr_filter_NotFoundTablesPath
-    COMMAND      ${CMAKE_BINARY_DIR}/bin/emoslib_bufr_filter
+    COMMAND      ${CMAKE_INSTALL_PREFIX}/bin/emoslib_bufr_filter
     ARGS         -r "${data}/request" -i "${data}/CRYO00012015020412.DC1.DAT" -o /dev/null
     ENVIRONMENT  "BUFR_TABLES=NotFoundTablesPath/" )
 set_tests_properties(
@@ -48,17 +48,17 @@ set_tests_properties(
 # unit tests: bufr_filter
 ecbuild_add_test(
     TARGET       emoslib_bufr_filter_NotFoundTestFile
-    COMMAND      ${CMAKE_BINARY_DIR}/bin/emoslib_bufr_filter
+    COMMAND      ${CMAKE_INSTALL_PREFIX}/bin/emoslib_bufr_filter
     ARGS         -r "${data}/request" -i "NotFoundTestFile" -o /dev/null
     ENVIRONMENT  "BUFR_TABLES=${TEST_BUFR_TABLES_PATH}" )
 ecbuild_add_test(
     TARGET       emoslib_bufr_filter_NotFoundRequest
-    COMMAND      ${CMAKE_BINARY_DIR}/bin/emoslib_bufr_filter
+    COMMAND      ${CMAKE_INSTALL_PREFIX}/bin/emoslib_bufr_filter
     ARGS         -r "NotFoundRequest" -i "${data}/CRYO00012015020412.DC1.DAT" -o /dev/null
     ENVIRONMENT  "BUFR_TABLES=${TEST_BUFR_TABLES_PATH}" )
 ecbuild_add_test(
     TARGET       emoslib_bufr_filter_CRYO00012015020412.DC1.DAT
-    COMMAND      ${CMAKE_BINARY_DIR}/bin/emoslib_bufr_filter
+    COMMAND      ${CMAKE_INSTALL_PREFIX}/bin/emoslib_bufr_filter
     ARGS         -r "${data}/request" -i "${data}/CRYO00012015020412.DC1.DAT" -o /dev/null
     ENVIRONMENT  "BUFR_TABLES=${TEST_BUFR_TABLES_PATH}" )
 
@@ -76,7 +76,7 @@ foreach( bufr_file
         "tracks_15U_ITA_2014041012.det.bufr4" )
     ecbuild_add_test(
         TARGET       "bufr_decode_all_${bufr_file}"
-        COMMAND      ${CMAKE_BINARY_DIR}/bin/bufr_decode_all
+        COMMAND      ${CMAKE_INTALL_PREFIX}/bin/bufr_decode_all
         ARGS         -i "${data}/${bufr_file}"
         TEST_DEPENDS download_tests_bufr_src_files
         ENVIRONMENT  "BUFR_TABLES=${TEST_BUFR_TABLES_PATH}" )
@@ -139,16 +139,16 @@ if( FALSE )
     set( in  "${PROJECT_SOURCE_DIR}/tests/bufr/data/tracks_05W_PEIPAH_2014041000.fm" )
     set( ref "${PROJECT_SOURCE_DIR}/tests/bufr/data/tracks_05W_PEIPAH_2014041000.eps.bufr" )
     set( out                                       "tracks_05W_PEIPAH_2014041000.eps.bufr" )
-    ecbuild_add_test( TARGET tc_tracks_eps_${out}  DEPENDS tc_tracks_eps COMMAND ${CMAKE_BINARY_DIR}/bin/tc_tracks_eps ARGS    -i "${in}" -o "${out}"  ENVIRONMENT "BUFR_TABLES=${TEST_BUFR_TABLES_PATH}" )
-    ecbuild_add_test( TARGET tc_tracks_eps_${out}4 DEPENDS tc_tracks_eps COMMAND ${CMAKE_BINARY_DIR}/bin/tc_tracks_eps ARGS -4 -i "${in}" -o "${out}4" ENVIRONMENT "BUFR_TABLES=${TEST_BUFR_TABLES_PATH}" )
+    ecbuild_add_test( TARGET tc_tracks_eps_${out}  DEPENDS tc_tracks_eps COMMAND ${CMAKE_INSTALL_PREFIX}/bin/tc_tracks_eps ARGS    -i "${in}" -o "${out}"  ENVIRONMENT "BUFR_TABLES=${TEST_BUFR_TABLES_PATH}" )
+    ecbuild_add_test( TARGET tc_tracks_eps_${out}4 DEPENDS tc_tracks_eps COMMAND ${CMAKE_INSTALL_PREFIX}/bin/tc_tracks_eps ARGS -4 -i "${in}" -o "${out}4" ENVIRONMENT "BUFR_TABLES=${TEST_BUFR_TABLES_PATH}" )
     ecbuild_add_test( TARGET tc_tracks_eps_${out}_cmp  TEST_DEPENDS tc_tracks_eps_${out}  CONDITION CMD_CMP COMMAND ${CMD_CMP} ARGS "${ref}" "${out}"  )
     ecbuild_add_test( TARGET tc_tracks_eps_${out}4_cmp TEST_DEPENDS tc_tracks_eps_${out}4 CONDITION CMD_CMP COMMAND ${CMD_CMP} ARGS "${ref}" "${out}4" )
 
     set( in  "${PROJECT_SOURCE_DIR}/tests/bufr/data/tracks_15U_ITA_2014041012.fm" )
     set( ref "${PROJECT_SOURCE_DIR}/tests/bufr/data/tracks_15U_ITA_2014041012.det.bufr" )
     set( out                                       "tracks_15U_ITA_2014041012.det.bufr" )
-    ecbuild_add_test( TARGET tc_tracks_det_${out}  DEPENDS tc_tracks_det COMMAND ${CMAKE_BINARY_DIR}/bin/tc_tracks_det ARGS    -i "${in}" -o "${out}"  ENVIRONMENT "BUFR_TABLES=${TEST_BUFR_TABLES_PATH}" )
-    ecbuild_add_test( TARGET tc_tracks_det_${out}4 DEPENDS tc_tracks_det COMMAND ${CMAKE_BINARY_DIR}/bin/tc_tracks_det ARGS -4 -i "${in}" -o "${out}4" ENVIRONMENT "BUFR_TABLES=${TEST_BUFR_TABLES_PATH}" )
+    ecbuild_add_test( TARGET tc_tracks_det_${out}  DEPENDS tc_tracks_det COMMAND ${CMAKE_INSTALL_PREFIX}/bin/tc_tracks_det ARGS    -i "${in}" -o "${out}"  ENVIRONMENT "BUFR_TABLES=${TEST_BUFR_TABLES_PATH}" )
+    ecbuild_add_test( TARGET tc_tracks_det_${out}4 DEPENDS tc_tracks_det COMMAND ${CMAKE_INSTALL_PREFIX}/bin/tc_tracks_det ARGS -4 -i "${in}" -o "${out}4" ENVIRONMENT "BUFR_TABLES=${TEST_BUFR_TABLES_PATH}" )
     ecbuild_add_test( TARGET tc_tracks_det_${out}_cmp  TEST_DEPENDS tc_tracks_det_${out}  CONDITION CMD_CMP COMMAND ${CMD_CMP} ARGS "${ref}" "${out}"  )
     ecbuild_add_test( TARGET tc_tracks_det_${out}4_cmp TEST_DEPENDS tc_tracks_det_${out}4 CONDITION CMD_CMP COMMAND ${CMD_CMP} ARGS "${ref}" "${out}4" )
 
